pkg/services/ngalert/api/api_prometheus.go

```diff
@@ -38,7 +46,108 @@ func (srv PrometheusSrv) RouteGetAlertStatuses(c *models.ReqContext) response.Re
+		ruleResponse.DiscoveryBase.ErrorType = apiv1.ErrServer
+		return response.JSON(http.StatusInternalServerError, ruleResponse)
+	}
+
+	for _, groupId := range ruleGroupQuery.Result {
+		alertRuleQuery := ngmodels.ListRuleGroupAlertRulesQuery{OrgID: c.SignedInUser.OrgId, RuleGroup: groupId}
+		if err := srv.store.GetRuleGroupAlertRules(&alertRuleQuery); err != nil {
+			ruleResponse.DiscoveryBase.Status = "error"
+			ruleResponse.DiscoveryBase.Error = fmt.Sprintf("failure getting rules for group %s: %s", groupId, err.Error())
+			ruleResponse.DiscoveryBase.ErrorType = apiv1.ErrServer
+			return response.JSON(http.StatusInternalServerError, ruleResponse)
+		}
+
+		newGroup := &apimodels.RuleGroup{
+			Name:           groupId,
+			File:           "", // TODO: set this once we are setting it somewhere
+			LastEvaluation: time.Time{},
+			EvaluationTime: 0, // TODO: see if we are able to pass this along with evaluation results
+		}
+		for _, rule := range alertRuleQuery.Result {
+			instanceQuery := ngmodels.ListAlertInstancesQuery{
+				DefinitionOrgID: c.SignedInUser.OrgId,
+				DefinitionUID:   rule.UID,
+			}
+			if err := srv.store.ListAlertInstances(&instanceQuery); err != nil {
+				ruleResponse.DiscoveryBase.Status = "error"
+				ruleResponse.DiscoveryBase.Error = fmt.Sprintf("failure getting alerts for rule %s: %s", rule.UID, err.Error())
+				ruleResponse.DiscoveryBase.ErrorType = apiv1.ErrServer
+				return response.JSON(http.StatusInternalServerError, ruleResponse)
+			}
+
+			alertingRule := apimodels.AlertingRule{
+				State:       "inactive",
+				Name:        rule.Title,
+				Query:       "", // TODO: get this from parsing AlertRule.Data
+				Duration:    time.Duration(rule.For).Seconds(),
+				Annotations: rule.Annotations,
+			}
+
+			newRule := apimodels.Rule{
+				Name:           rule.Title,
+				Labels:         nil,                    // TODO: NG AlertRule does not have labels but does have annotations
+				Health:         "",                     // TODO: how is this determined?
+				Type:           apiv1.RuleTypeAlerting, // TODO: can we guarantee that only alerting rules will be returned by the DB?
+				LastEvaluation: time.Time{},
+				EvaluationTime: 0, // TODO: set this once we are saving it or adding it to evaluation results
+			}
+			for _, instance := range instanceQuery.Result {
+				activeAt := instance.CurrentStateSince
+				alert := &apimodels.Alert{
+					Labels:      ngmodels.ToMap(instance.Labels),
+					Annotations: nil, // TODO: set these once they are added to evaluation results
+					State:       translateInstanceState(instance.CurrentState),
+					ActiveAt:    &activeAt,
+					Value:       "", // TODO: set this once it is added to the evaluation results
+				}
+				if instance.LastEvalTime.After(newRule.LastEvaluation) {
+					newRule.LastEvaluation = instance.LastEvalTime
+					newGroup.LastEvaluation = instance.LastEvalTime
+				}
+				switch alert.State {
```

Comment:
Not sure I follow this line here, what's the difference between this state and `translateInstanceState`?

Reply:
translateInstanceState is just a quick mapping of the states as represented in the enum from instance.go. This is to make sure that the rule is set to the correct state in the event that any instances are not in inactive state.